Fix navigation link popover staying open after creating page

jeryj opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
<!-- Link this PR to its associated issue with an appropriate keyword: Closes, See, Follow up to, etc. -->
Closes https://github.com/WordPress/gutenberg/issues/73861
Fix focus loss after creating page in navigation block. This focus loss means the close on focus outside handler never gets called, so the popover can't be closed.

<!-- In a few words, what is the PR actually doing? -->
## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->
Fixes a bug :)

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->
1. Only skip auto-focus on mount when entering edit mode, not preview mode. This ensures focus is properly placed in the link preview after creating a page through the navigation block's "Create page" dialog.
2. In the link-ui for the navigation block, force editing to be false after creating a new page. Otherwise, e2e tests will catch that there is a brief transition to the edit state while link values are being updated.


## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a post or page. -->
<!-- 2. Insert a heading block. -->
<!-- 3. etc. -->
- Click the navigation link appender
- Click Add page in the popover
- Add a page title
- Click Create page
- You should see the link preview for your new page
- Click somewhere else on the page to close the popover
- Popover should close

Additional coverage: Copy the contents of the e2e test into trunk and verify it fails on trunk and passes on this branch: `npm run test:e2e test/e2e/specs/editor/blocks/navigation.spec.js -- -g "Can create a new page using the navigation block appender"`

### Testing Instructions for Keyboard
<!-- How can you test the changes by using the keyboard only? Please note, this is required for PRs that change the user interface (UI). This ensures the PR can be tested for any possible accessibility regressions. -->
- navigate to the navigation link appender using your tab key
- Press enter
- Tab to Add page button
- Tab to the page title field and enter a title
- Tab to Create page
- You should see the link preview for your new page
- Press escape
- Popover should close
- Focus should be on the appender
## Screenshots or screencast <!-- if applicable -->

<!-- If you would like to upload screenshots, feel free to use the table below when it is useful to show the difference between before and after the change. -->

|Before|After|
|-|-|
|<!-- Before screenshot here -->|<!-- After screenshot here -->|

jeryj committed:
Fix focus loss after creating page in navigation block

Only skip auto-focus on mount when entering edit mode, not preview mount. This ensures focus is properly placed in the link preview after creating a page through the navigation block's "Create page" dialog.

jeryj committed:
Switch to preview after creating page in via navigation link appender

This is necessary for e2e tests, as there is a brief transition where the page is created but the link control wants to be in edit mode while values are being updated.

jeryj committed:
Add e2e test for bug

tellthemachines reviewed (commented):
Thanks for the quick PR @jeryj ! While it's working fine to fix the reported issue, the e2e failures in `test/e2e/specs/editor/blocks/links.spec.js` seem legit so this is probably breaking something elsewhere 😅

jeryj committed:
Fix focus management after creating new page in navigation block link ui

The previous LinkControl fix for this was too heavy handed and caused issues with always moving focus into the preview when mounting the link control. I've reverted that and used a more targeted bug fix approach by setting a flag that we need to send focus into the link control after creating a new page.

jeryj committed:
Refactor focus management into one flow for link ui

jeryj committed:
Add test steps for checking focus management

jeryj committed:
Focus first tabbable instead of first focusable so we don't include the wrapper

tellthemachines reviewed (approved):
LGTM!
